discussment-core/src/main/java/org/danekja/discussment/core/dao/jpa/GenericDaoJPA.java

```diff
@@ -40,12 +43,15 @@ public T getById(PK id) {
     }
 
     public void remove(T obj) {
-        if(!em.getTransaction().isActive()) {
+        boolean isSpringActive = em.isJoinedToTransaction();
```

Comment:
Doesnt necessarily mean Spring is active, simply checks for existing transaction. Rename to hasOuterTransaction